Minor improvements

wellme opened the pull request:
This is pretty much just the list of changes from my original branch (Its pull request should be closed)

**Port selection**
Requested by some random loser in discord. Port is specified by using the spinner, but it defaults to 25525.
When a client attempts to connect to a server, it must specify the port in the TextInputBox with this format: IPv4:port (ex: 192.168.1.107:25526)
If the port is the default port (25525), then it can be omitted (i.e. 192.168.1.107 is interpreted as 192.168.1.107:25525).
Errors while parsing the port results in the usage of the default port (i.e. 192.168.1.107:owo is interpreted as 192.168.1.107:25525).

**Clipboard support**
The server's running panel now includes a "Copy to clipboard" button, which copies the IP (and port, if applicable) of the server.
The client _should_ be able to copy/paste using ctrl+c and ctrl+v respectively. Copying copies the whole input of the text box (because selection is not a thing in this component) and pasting inserts whatever text is in the clipboard. Invalid characters, such as newlines, foreign letters (éàèî, etc) are removed from the string.

**Code related**
Moved the Swing components of the FEServer class to their own files in net.fe.network.serverui

SapphireTactician commented:
@wellme  My compiler caught fire, please advise.

```
./src/net/fe/modifier/MadeInChina.java:28: error: illegal start of expression
		return limits.copyWithNewFunds((i) -> i * 2);
		                                    ^
./src/net/fe/builderStage/WaitStage.java:67: error: ')' expected
					net.fe.unit.UnitFactory::getUnit,
					                       ^
./src/net/fe/builderStage/WaitStage.java:67: error: illegal start of expression
					net.fe.unit.UnitFactory::getUnit,
					                        ^
./src/net/fe/builderStage/WaitStage.java:68: error: ';' expected
					net.fe.unit.Item.getAllItems(),
					   ^
./src/net/fe/builderStage/WaitStage.java:68: error: ';' expected
					net.fe.unit.Item.getAllItems(),
					                              ^
./src/net/fe/builderStage/WaitStage.java:69: error: ';' expected
					session.getModifiers()
					                      ^
./src/net/fe/unit/Item.java:113: error: ')' expected
		WeaponFactory.getAllWeapons().forEach(retVal::add);
		                                            ^
./src/net/fe/unit/Item.java:113: error: illegal start of expression
		WeaponFactory.getAllWeapons().forEach(retVal::add);
		                                             ^
./src/net/fe/unit/Item.java:113: error: ';' expected
		WeaponFactory.getAllWeapons().forEach(retVal::add);
		                                                 ^
./src/net/fe/unit/Unit.java:388: error: illegal start of expression
		this.skills.forEach((s) -> copy.addSkill(s.getCopy()));
		                         ^
./src/net/fe/unit/Class.java:68: error: illegal start of expression
				this.usableWeapon.stream().mapToInt((x) -> 1 << x.ordinal()).sum();
				                                         ^
./src/net/fe/FEMultiplayer.java:215: error: ')' expected
		CombatCalculator calc = new CombatCalculator(new UnitIdentifier(u1), new UnitIdentifier(u2), FEMultiplayer::getUnit);
		                                                                                                          ^
./src/net/fe/FEMultiplayer.java:215: error: illegal start of expression
		CombatCalculator calc = new CombatCalculator(new UnitIdentifier(u1), new UnitIdentifier(u2), FEMultiplayer::getUnit);
		                                                                                                           ^
./src/net/fe/FEMultiplayer.java:215: error: ';' expected
		CombatCalculator calc = new CombatCalculator(new UnitIdentifier(u1), new UnitIdentifier(u2), FEMultiplayer::getUnit);
		                                                                                                                   ^
./src/net/fe/network/serverui/FEServerFrame.java:36: error: ')' expected
		mainPanel.setServerStartRunnable(this::serverStart);
		                                     ^
./src/net/fe/network/serverui/FEServerFrame.java:36: error: illegal start of expression
		mainPanel.setServerStartRunnable(this::serverStart);
		                                      ^
./src/net/fe/network/serverui/FEServerFrame.java:36: error: ';' expected
		mainPanel.setServerStartRunnable(this::serverStart);
		                                                  ^
./src/net/fe/network/serverui/FEServerMainPanel.java:182: error: illegal start of expression
		modifiersList.setToolTipFunction(modifier -> modifier.getDescription());
		                                           ^
./src/net/fe/network/serverui/FEServerMainPanel.java:189: error: illegal start of expression
		selectedModifiersList.setToolTipFunction(modifier -> modifier.getDescription());
		                                                   ^
./src/net/fe/network/serverui/FEServerRunPanel.java:80: error: illegal start of expression
		kickAll.addActionListener((e2) -> FEServer.resetToLobbyAndKickPlayers());
		                                ^
./src/net/fe/network/command/AttackCommand.java:36: error: illegal start of expression
		CombatCalculator calc = new CombatCalculator(unitId, otherId, (ui) -> stage.getUnit(ui));
		                                                                    ^
./src/net/fe/network/command/HealCommand.java:36: error: illegal start of expression
		HealCalculator calc = new HealCalculator(unitId, otherId, (ui) -> stage.getUnit(ui));
		                                                                ^
./src/net/fe/network/message/PartyMessage.java:70: error: ')' expected
		realItems.forEach(realItems2::add);
		                            ^
./src/net/fe/network/message/PartyMessage.java:70: error: illegal start of expression
		realItems.forEach(realItems2::add);
		                             ^
./src/net/fe/network/message/PartyMessage.java:70: error: ';' expected
		realItems.forEach(realItems2::add);
		                                 ^
./src/net/fe/network/message/PartyMessage.java:77: error: illegal start of expression
				usedRes = usedRes.copyWithNewFunds((i) -> i + u2.squeezeGold());
				                                        ^
./src/net/fe/network/message/PartyMessage.java:78: error: illegal start of expression
				usedRes = usedRes.copyWithNewExp((i) -> i + u2.squeezeExp());
				                                      ^
./src/net/fe/overworldStage/ClientOverworldStage.java:144: error: ')' expected
					.filter((Unit u) -> u.getHp() > 0)
					             ^
./src/net/fe/overworldStage/ClientOverworldStage.java:144: error: illegal start of expression
					.filter((Unit u) -> u.getHp() > 0)
					                  ^
./src/net/fe/overworldStage/ClientOverworldStage.java:144: error: ';' expected
					.filter((Unit u) -> u.getHp() > 0)
					                                 ^
./src/net/fe/overworldStage/ClientOverworldStage.java:145: error: ')' expected
					.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
					          ^
./src/net/fe/overworldStage/ClientOverworldStage.java:145: error: illegal start of expression
					.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
					               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:145: error: ';' expected
					.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
					                                                       ^
./src/net/fe/overworldStage/ClientOverworldStage.java:146: error: '.class' expected
					.toArray(Node[]::new);
					               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:146: error: ';' expected
					.toArray(Node[]::new);
					                ^
./src/net/fe/overworldStage/ClientOverworldStage.java:146: error: <identifier> expected
					.toArray(Node[]::new);
					                    ^
./src/net/fe/overworldStage/ClientOverworldStage.java:278: error: ')' expected
		messages.forEach(pendingMessages::add);
		                                ^
./src/net/fe/overworldStage/ClientOverworldStage.java:278: error: illegal start of expression
		messages.forEach(pendingMessages::add);
		                                 ^
./src/net/fe/overworldStage/ClientOverworldStage.java:278: error: ';' expected
		messages.forEach(pendingMessages::add);
		                                     ^
./src/net/fe/overworldStage/ClientOverworldStage.java:431: error: ')' expected
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						             ^
./src/net/fe/overworldStage/ClientOverworldStage.java:431: error: illegal start of expression
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						                  ^
./src/net/fe/overworldStage/ClientOverworldStage.java:431: error: ';' expected
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						                                                   ^
./src/net/fe/overworldStage/ClientOverworldStage.java:432: error: ')' expected
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						          ^
./src/net/fe/overworldStage/ClientOverworldStage.java:432: error: illegal start of expression
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:432: error: ';' expected
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						                                                       ^
./src/net/fe/overworldStage/ClientOverworldStage.java:433: error: '.class' expected
						.toArray(Node[]::new);
						               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:433: error: ';' expected
						.toArray(Node[]::new);
						                ^
./src/net/fe/overworldStage/ClientOverworldStage.java:433: error: <identifier> expected
						.toArray(Node[]::new);
						                    ^
./src/net/fe/overworldStage/ClientOverworldStage.java:449: error: ')' expected
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						             ^
./src/net/fe/overworldStage/ClientOverworldStage.java:449: error: illegal start of expression
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						                  ^
./src/net/fe/overworldStage/ClientOverworldStage.java:449: error: ';' expected
						.filter((Unit u) -> u.getHp() > 0 && !u.isRescued())
						                                                   ^
./src/net/fe/overworldStage/ClientOverworldStage.java:450: error: ')' expected
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						          ^
./src/net/fe/overworldStage/ClientOverworldStage.java:450: error: illegal start of expression
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:450: error: ';' expected
						.map((Unit u) -> new Node(u.getXCoord(), u.getYCoord()))
						                                                       ^
./src/net/fe/overworldStage/ClientOverworldStage.java:451: error: '.class' expected
						.toArray(Node[]::new);
						               ^
./src/net/fe/overworldStage/ClientOverworldStage.java:451: error: ';' expected
						.toArray(Node[]::new);
						                ^
./src/net/fe/overworldStage/ClientOverworldStage.java:451: error: <identifier> expected
						.toArray(Node[]::new);
						                    ^
./src/net/fe/overworldStage/OverworldStage.java:272: error: illegal start of expression
					((EndTurn) message).checkHp((ui) -> this.getUnit(ui));
					                                  ^
58 errors
```

SapphireTactician commented:
@wellme @eliatlarge I am ever so sorry for taking as long as I did. Life and BoTW happened.

Anyways, I think it's safe to merge this based on what I've done. However, idk if you think it should be in it's own test branch until more of the community can play around with it since I feel any changes to the networking code should be tested by more than just me and my few guinea pigs. I'd like to post this as a release so the community can download a binary of it, but I can't do that until you, @eliatlarge , merge it or give it a test branch somehow.

wellme commented:
I can expand the text box for the address to this:
![image](https://cloud.githubusercontent.com/assets/7041069/25531304/3dcad656-2bf7-11e7-85fc-58a4962e731f.png)
- Made the text box larger
- Moved the "go" button to the right
- Changed "Server IP" to "Server address"

Unfortunately, based on how the buttons are implemented (i.e. using a static image), I can't do anything fancy without destroying the aspect ratio of the text.

SapphireTactician commented:
@wellme That looks perfect to me. 👍 Make it happen. :)

wellme commented:
Added two more commits. The first prevents freezes when a client tries to connect to a server. The second changes nothing, it's just code formatting.

rayrobdod commented on src/net/fe/network/serverui/FEServerRunPanel.java:
```diff
@@ -48,56 +42,53 @@ public FEServerRunPanel(int port) {
 	}
-	
+
 	public void setPort(int port) {
```
If you're going to remove the `port` variable, you might as well remove its setter too.

Jonje90 commented:
What can I do to help?

eliatlarge commented:
@Jonje90 just go ahead and look at the issues page and see what you can do. I also recommend talking to us on our subreddit at reddit.com/r/femp and on our discord, the link being in the sidebar. hope to see you!

Jonje90 commented:
@eliatlarge haha....one problem. I know this was pretty retarded of me, but... I have no idea how to do any of this (coding, that is). Sorry, but could you point me to a guide on coding, Java, etc., or something? I just saw this whole thing and felt like I wanted to help. sorry again

eliatlarge commented:
@Jonje90 This isn't really the best place to discuss this, as this is a pull request thread, though your willingness to help is very much appreciated! If you would like to continue discussing this, I recommend you email me at eliatlarge@gmail.com if you have any more questions.

I personally recommend codecademy.com to learn Java, however, as this is a quite old and convoluted project, you might have difficulty getting into development for it.

For FEMP itself, I wrote a few tutorials for getting started:

- http://imgur.com/gallery/7rCOp/

- http://imgur.com/gallery/2c0zs

- http://imgur.com/gallery/LKtdO/

However, if none of this seems like something you can see yourself doing, we regularly talk about changes and balancing on our discord. We also can use help setting up events and hosting servers if that's your cup of tea, so that would be cool too.

eliatlarge commented:
closing to merge to testbranch this time
